src/Magnum/ImGuiIntegration/Integration.cpp

```diff
@@ -0,0 +1,264 @@
+    io.KeyMap[ImGuiKey_Z]          = ImGuiKey_Z;
+
+    //   io.SetClipboardTextFn = ;
+    //   io.GetClipboardTextFn = ;
+
+    unsigned char *pixels;
+    int width, height;
+    int pixelSize;
+    io.Fonts->GetTexDataAsRGBA32(&pixels, &width, &height, &pixelSize);
+
+    CORRADE_ASSERT(width > 0,
+            "ImGuiIntegration::Context::Context(): Expected width to be larger than 0", );
+    CORRADE_ASSERT(height > 0,
+            "ImGuiIntegration::Context::Context(): Expected height to be larger than 0", );
+    CORRADE_ASSERT(pixelSize > 0,
+            "ImGuiIntegration::Context::Context(): Expected pixel size to be larger than 0", );
+
+    ImageView2D image{GL::PixelFormat::RGBA,
+        GL::PixelType::UnsignedByte, {width, height},
+        {pixels, std::size_t(pixelSize*width*height)}};
+
+    _texture.setMagnificationFilter(GL::SamplerFilter::Linear)
+        .setMinificationFilter(GL::SamplerFilter::Linear)
+        .setStorage(1, GL::TextureFormat::RGBA8, image.size())
+        .setSubImage(0, {}, image);
+
+    _mesh.setPrimitive(GL::MeshPrimitive::Triangles);
+    _mesh.addVertexBuffer(
+            _vertexBuffer, 0, Implementation::ImGuiShader::Position{},
+            Implementation::ImGuiShader::TextureCoordinates{},
+            Implementation::ImGuiShader::Color(Implementation::ImGuiShader::Color::Components::Four,
+                Implementation::ImGuiShader::Color::DataType::UnsignedByte,
+                Implementation::ImGuiShader::Color::DataOption::Normalized));
+
+    _timeline.start();
+
+    _instance = this;
+}
+
+Context::~Context() {
+    /* Not initialized/initialization failed */
+    if(!_instance) return;
+
+    ImGui::DestroyContext();
+
+    _instance = nullptr;
+}
+
+void Context::newFrame(const Vector2i &winSize, const Vector2i &viewportSize) {
+    _timeline.nextFrame();
+
+    ImGuiIO& io = ImGui::GetIO();
+
+    /* Setup display size (every frame to accommodate for window resizing) */
+    io.DisplaySize = ImVec2(Vector2(winSize));
+    io.DisplayFramebufferScale =
+        ImVec2(winSize.x() > 0 ? float(viewportSize.x())/winSize.x() : 0.0f,
+               winSize.y() > 0 ? float(viewportSize.y())/winSize.y() : 0.0f);
+
+    io.DeltaTime = _timeline.previousFrameDuration();
+
+    ImGui::NewFrame();
+}
+
+void Context::drawFrame() {
+    ImGui::Render();
+
+    ImGuiIO& io = ImGui::GetIO();
+    const int fbWidth = int(io.DisplaySize.x*io.DisplayFramebufferScale.x);
+    const int fbHeight = int(io.DisplaySize.y*io.DisplayFramebufferScale.y);
+    if(fbWidth == 0 || fbHeight == 0)
+        return;
+
+    auto drawData = ImGui::GetDrawData();
+    if(!drawData)
+        return;
+
+    drawData->ScaleClipRects(io.DisplayFramebufferScale);
+
+    const Matrix4 projection{
+        {2.0f/io.DisplaySize.x, 0.0f, 0.0f, 0.0f},
+        {0.0f, -2.0f/io.DisplaySize.y, 0.0f, 0.0f},
+        {0.0f, 0.0f, -1.0f, 0.0f},
+        {-1.0f, 1.0f, 0.0f, 1.0f},
+    };
+    _shader.setProjectionMatrix(projection);
```

Comment:
Not really sure, but maybe composing the matrix from primitive operations ("move origin to top left", "switch to Y down", "projection the size of the display") would make it appear a bit more self-documenting?